src/main/java/Reporter.java

```diff
@@ -117,29 +132,56 @@ private void reportSprite(int spriteNo, String spriteName, PrintWriter pw, Sb2 s
         if (shouldReport(SPRITE_HEADERS)) {
             pw.write("\n" + tab + "Sprite " + spriteNo + ": " + spriteName + "\n");
         }
-        int[] scriptLengths = sb2.getScriptLengthsForSprite(spriteName);
+        Script[] scripts = sb2.getScriptsForSprite(spriteName);
         if (shouldReport(SCRIPTS_PER_SPRITE)) {
-            pw.write(tab + scriptLengths.length + " scripts\n");
+            pw.write(tab + scripts.length + " scripts\n");
         }
-        for (int i = 0; i < scriptLengths.length; i++) {
-            reportScript(i + 1, scriptLengths[i], pw);
+        for (int i = 0; i < scripts.length; i++) {
+            reportScript(i + 1, scripts[i], pw);
         }
     }
     /**
      * Report one Script from a Sprite project.
      * @param scriptNo The index of the Script in the list.  Starting from 1.
-     * @param scriptLength The length of the script.
+     * @param script The script to report.
      * @param pw The PrintWriter.
      */
-    private void reportScript(int scriptNo, int scriptLength, PrintWriter pw) {
+    private void reportScript(int scriptNo, Script script, PrintWriter pw) {
         String tab = TAB + TAB;
         if (shouldReport(SCRIPT_HEADERS)) {
             pw.write(tab + "Script " + scriptNo + "\n");
         }
         tab = tab + TAB;
         if (shouldReport(SCRIPT_LENGTHS)) {
-            pw.write(tab + "length = " + scriptLength + "\n");
+            pw.write(tab + "length = " + script.getLength() + "\n");
+        }
+        if (shouldReport(SCRIPT_BYCATTOT)) {
+            pw.write(tab + "blocks by category: "
+                     + blocksByCatString(script.sumBlocksByCategory()) + "\n");
         }
     }
-
+    /**
+     * Produces a nice string representeing blocks by category.
+     * @param blocksByCat Array containing block counts by category
+     * @return The nice string
+     */
+    private String blocksByCatString(int[] blocksByCat) {
+        String[] cats = ScriptSpecs.getCategories();
```

Comment:
Can we all take a moment to enjoy that you got to use `cats` as a variable name? I find this immensely pleasing.